site/content/blog/identify-ghost-diff-during-kc-reconcile.md

```diff
@@ -0,0 +1,274 @@
+$ kubectl get pkg simple-app.corp.com.1.0.0 -oyaml > copy-simple-app-package.yaml
+```
+
+Open copy-simple-app-package.yaml. Remove labels starting with `Kapp`. Add the below snippet to the Kapp section. Setting the `diff-changes` to true will enable the `Kapp` to show changes.
+```
+...
+- kapp:
+    rawOptions:
+    - --diff-changes=true
+...
+```
+
+I would recommend not to tinker with the original package. Hence, let's change the Package version. Lets change it from 1.0.0 to 2.0.0. Specifically, change the `spec.Version` and `metadata.labels.name`. Now, apply this Package in the cluster so that it will be available for install.
+
+```bash 
+$ kubectl apply -f copy-simple-app-package.yaml
+package.data.packaging.carvel.dev/simple-app.corp.com.2.0.0 created
+```
+
+Now, if we will see list of available Packages, we can see our locally created Package as well. 
+
+```bash
+$ kctrl package available list --summary=false
+Target cluster 'https://192.168.64.82:8443' (nodes: minikube)
+
+Available packages in namespace 'default'
+
+Name                 Version  Released at
+simple-app.corp.com  1.0.0    0001-01-01 00:00:00 +0000 UTC
+simple-app.corp.com  2.0.0    0001-01-01 00:00:00 +0000 UTC
+
+Succeeded
+```
+
+Let's uninstall the previous package and deploy locally created package.
+
+```bash
+$ kctrl package installed delete -i pkg-demo -y
+
+Delete package install 'pkg-demo' from namespace 'default'
+Target cluster 'https://192.168.64.82:8443' (nodes: minikube)
+Deleting package install 'pkg-demo' from namespace 'default'
+Waiting for deletion of package install 'pkg-demo' from namespace 'default'
+
+2:08:42PM: packageinstall/pkg-demo (packaging.carvel.dev/v1alpha1) namespace: default: Deleting
+2:08:43PM: packageinstall/pkg-demo (packaging.carvel.dev/v1alpha1) namespace: default: DeletionSucceeded
+
+Deleting 'ClusterRole': pkg-demo-default-cluster-role
+Deleting 'ClusterRoleBinding': pkg-demo-default-cluster-rolebinding
+Deleting 'ServiceAccount': pkg-demo-default-sa
+
+Succeeded
+
+$ kctrl package install -i pkg-demo -p simple-app.corp.com --version 2.0.0
+
+Target cluster 'https://192.168.64.82:8443' (nodes: minikube)
+Creating service account 'pkg-demo-default-sa'
+Creating cluster admin role 'pkg-demo-default-cluster-role'
+Creating cluster role binding 'pkg-demo-default-cluster-rolebinding'
+Creating package install resource
+Waiting for PackageInstall reconciliation for 'pkg-demo'
+
+2:12:19PM: packageinstall/pkg-demo (packaging.carvel.dev/v1alpha1) namespace: default: Reconciling
+2:12:28PM: packageinstall/pkg-demo (packaging.carvel.dev/v1alpha1) namespace: default: ReconcileSucceeded
+
+Succeeded
+```
+After the Package is deployed successfully, let's see what the initial configuration of the resources looks like. We can get that by describing App(link to the app section) linked to the Package. Similar to configmap, the package creates `App` with the same name as its name. As the output is long, I have added only a small snippet.
```

Comment:
> App(link to the app section)

Looks like you put this for reminder and later forgot to put the link.  +1 for having a link here.

Reply:
Adopted